tooling/metamodel-generator/src/main/java/org/hibernate/processor/util/GenericTypeParameterResolver.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.processor.util;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeMirror;
+import java.util.HashMap;
+import java.util.Map;
+
+public class GenericTypeParameterResolver {
+
+	private final Map<Element, Map<String, TypeMirror>> typeParameterMap;
```

Comment:
Given that I don't think we can hold onto these in a `static` (see other comment), I guess I would prefer to not even build up this map here. This search can be stateless, I believe (unless we're concerned we might encounter circularities, but I guess I doubt we would).